contrastscan-remote/src/main/java/com/contrastsecurity/scan/ContrastRemoteScanner.kt

```diff
@@ -0,0 +1,193 @@
+package com.contrastsecurity.scan
+
+import com.contrastsecurity.exceptions.HttpResponseException
+import com.contrastsecurity.sdk.ContrastSDK
+import com.contrastsecurity.sdk.UserAgentProduct
+import com.contrastsecurity.sdk.scan.CodeArtifact
+import com.contrastsecurity.sdk.scan.Project
+import com.contrastsecurity.sdk.scan.Scan
+import com.contrastsecurity.sdk.scan.ScanSummary
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.io.UncheckedIOException
+import java.net.MalformedURLException
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.time.Duration
+import java.util.concurrent.*
+import java.util.function.Consumer
+import kotlin.io.path.*
+
+/**
+ * A container to hold Contrast credential data.
+ */
+data class ContrastPlatformCredentials(
+    val user: String,
+    val orgId: String,
+    val apikey: String,
+    val serviceKey: String);
+
+class ContrastRemoteScanner(private val creds: ContrastPlatformCredentials, val apiUrl: URL, val projectName: String, val labelName: String) {
+    val sdk: ContrastSDK = connectToContrast(creds, apiUrl)
+
+    fun scan(prescan: Path, artifact: Path, scanOutputPath: Path, timeoutMs: Long) {
+        val project: Project = findOrCreateProject()
+
+        logger.info("Uploading " + artifact.fileName + " to Contrast Scan")
+
+        val codeArtifact: CodeArtifact
+        codeArtifact = try {
+            project.codeArtifacts().upload(artifact, prescan)
+        } catch (e: IOException) {
+            throw RuntimeException("Failed to upload code artifact to Contrast Scan", e)
+        } catch (e: HttpResponseException) {
+            throw RuntimeException("Failed to upload code artifact to Contrast Scan", e)
```

Comment:
all exceptions are unchecked in Kotlin, so there's no need to do this weird wrapping in RuntimeException we know from Java